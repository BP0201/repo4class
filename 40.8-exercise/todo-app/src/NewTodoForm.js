import React, { useState } from 'react';

const NewTodoForm = ({ addTodo }) => {
    const initialState = {
        todo: ""
    }

    const handleChange = e => {
        const { name, value } = e.target
        setFormData(data => ({
            ...data,
            [name]: value
        }))
    }

    const handleSubmit = e => {
        e.preventDefault();
        addTodo(formData)
        setFormData(initialState)
    }

    const [formData, setFormData] = useState(initialState)
    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="todo">Todo</label>
            <input 
                id="todo"
                name="todo"
                type="text"
                value={formData.todo}
                onChange={handleChange}
            />
            <button>Add Todo!</button>
        </form>
    )
}

export default NewTodoForm;