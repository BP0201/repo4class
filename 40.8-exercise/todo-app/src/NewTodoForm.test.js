import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

test('renders without crashing', () => {
  render(<NewTodoForm />);
});

test("it matches the snapshot", () => {
  const { asFragment } = render(<NewTodoForm />)
  expect(asFragment()).toMatchSnapshot();
})