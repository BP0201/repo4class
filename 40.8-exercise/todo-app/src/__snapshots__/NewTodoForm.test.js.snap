// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      Todo
    </label>
    <input
      id="todo"
      name="todo"
      type="text"
      value=""
    />
    <button>
      Add Todo!
    </button>
  </form>
</DocumentFragment>
`;
